src/core/execute.c

```diff
@@ -501,7 +516,7 @@ int exec_spawn(
         log_command_line(unit, "About to execute", command->path, command->argv);
 
         if (params->cgroup_path) {
-                r = exec_params_get_cgroup_path(params, cgroup_context, &subcgroup_path);
+                r = exec_params_get_cgroup_path(params, cgroup_context, params->cgroup_path, &subcgroup_path);
                 if (r < 0)
                         return log_unit_error_errno(unit, r, "Failed to acquire subcgroup path: %m");
                 if (r > 0) {
```

Comment:
So, hmm, I think with changes here it makes more sense to create delegate_subgroup only in exec_invoke, and not here, since we're no longer spawning into subcg anymore, and the rule is temporarily violated.